README.md

```diff
@@ -1 +1,83 @@
-# next-compression-plugin
\ No newline at end of file
+# next-compression-plugin
+
+This plugin compresses and serves previously compressed files using **NextJS** and **Express**.
+
+## Installation
+
+```sh
+$ npm i --save next-compression-plugin
+```
+
+## Usage
+
+### next.config.js  
+
+NextJS plugin used to compress files.
+
+```javascript
+const { withCompression } = require('next-compression-plugin');
+
+module.exports = withCompression({ ...nextConfig });
+```
+
+**Note**: At this time withCompression does not support configurable options.
+
+### Express
+Express middleware used to serve the previously compressed files. This package uses the [express-static-gzip](https://www.npmjs.com/package/express-static-gzip).
+
+```javascript
+//server.js
+const { compressionPlugin } = require('next-compression-plugin');
+
+const server = express();
+
+const options = {
+    requestPath: '/',
+    fsPath: '/directory/static'
+    serveStatic: {
+        maxAge: 365 * 24 * 60 * 60 * 1000, // 1 year
+        immutable: true,
+    }
+} 
+
+server.use(compressionPlugin(options));
+```
+
+#### Available options
+* **requestPath**: string 
```

Comment:
I think we can improve the formatting of options, either:

- use something similar to https://github.com/moxystudio/next-runtime-env#api
- use something similar to https://github.com/sindresorhus/p-retry#options